src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/PsiProvider.kt

```diff
@@ -0,0 +1,64 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.openapi.vfs.VfsUtilCore
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiManager
+import com.intellij.psi.util.PsiTreeUtil
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtTreeVisitorVoid
+
+class PsiProvider {
+
+    fun extractMethodsFromProject(project: Project): MutableList<KtNamedFunction> {
+        val ktMethods = mutableListOf<KtNamedFunction>()
+        val ktFiles = extractPsiFiles(project)
+        ktFiles.forEach { ktFile ->
+            collectPsiMethods(ktFile).forEach { function ->
+                ktMethods.add(function)
+            }
+        }
+        return ktMethods
+    }
```

Comment:
You can use: 
```
return extractPsiFiles(project).map { collectPsiMethods(it) }.flatten().toMutableList()
```
or simple:
```
return extractPsiFiles(project).map { collectPsiMethods(it) }.flatten()
```
and change `MutableList` to `List` in the return value type